navigation/ios/NavigationState.swift

```diff
@@ -49,18 +49,22 @@ class NavigationState<
         actionListener.startListening()
     }
 
-    override func push(route: PilotNavigationRoute) -> Bool {
+    override func canNavigate(route: PilotNavigationRoute) -> Bool {
+        guard let buildNavigation else { return false }
+        guard let route = route as? Route else { return false }
+        guard (buildNavigation(currentStack(), route) != nil) else { return false }
```

Comment:
This is ok to do that on iOS, because calling buildNavigation will build the ViewModel so you will always build a new instance of the VM for nothing 😬 You will need another method "canNavigate" on iOS side to only do the check without building the VM.

Reply:
so the check would only be on
guard let buildNavigation else { return false }
guard let route = route as? Route else { return false }?